pvlib/solarposition.py

```diff
@@ -1445,3 +1453,80 @@ def sun_rise_set_transit_geometric(times, latitude, longitude, declination,
     sunset = _local_times_from_hours_since_midnight(times, sunset_hour)
     transit = _local_times_from_hours_since_midnight(times, transit_hour)
     return sunrise, sunset, transit
+
+
+def spencer(times, latitude, longitude):
+    """
+    Calculate the solar position using a formulation by Spencer 1971/1972.
+
+    Parameters
+    ----------
+    times : pandas.DatetimeIndex`
+        Corresponding timestamps, must be localized to the timezone for the
+        ``latitude`` and ``longitude``.
+    latitude : float
+        Latitude in degrees, positive north of equator, negative to south
+    longitude : float
+        Longitude in degrees, positive east of prime meridian, negative to west
+
+    Returns
+    -------
+    DataFrame
+        The DataFrame will have the following columns:
+        zenith (degrees),
+        elevation (degrees),
+        azimuth (degrees),
+        equation_of_time (seconds),
+        declination (degrees).
+
+    References
+    ----------
+    [1] Spencer (1972) can be found in
+    Sen, Zekai. Solar energy fundamentals and modeling techniques:
+     atmosphere, environment, climate change and renewable energy.
+     Springer Science & Business Media, 2008.
+    """
+
+    julians, julian_2000 = datetime_to_julian(times)
```

Comment:
@meteoDaniel can you please replace all of these lines
https://github.com/pvlib/pvlib-python/blob/03bb14384d896b45b485ea88d70521f878d1e23a/pvlib/solarposition.py#L1490-L1511
with a call to `solar_zenith_analytical(latitude, hourangle, declination)` ?